immer/algorithm.hpp

```diff
@@ -153,8 +153,8 @@ T accumulate(Iterator first, Iterator last, T init, Fn fn)
  */
 template <typename Iterator, typename Sentinel,
           std::enable_if_t
-	  <detail::std_distance_supports_v<Iterator,Sentinel>, bool> = true>
-std::size_t distance(Iterator first, Sentinel last)
+          <detail::std_distance_supports_v<Iterator,Sentinel>, bool> = true>
+auto distance(Iterator first, Sentinel last)
```

Comment:
The "Almost Always use Auto" is about variables, not about return types. Sorry if I confused you about that. In this particular case I actually prefer the old version :sweat_smile:

Reply:
No problem. Changes incoming.